layer/filestore.go

```diff
@@ -90,6 +115,29 @@ func (fms *fileMetadataStore) StartTransaction(cacheID string) (*fileMetadataTra
 	return tx, nil
 }
 
+func (fms *fileMetadataStore) ListExistingTransactions() ([]fileMetadataTxData, error) {
+	tmpDir := filepath.Join(fms.root, "tmp")
+	if stat, err := os.Stat(tmpDir); err != nil {
```

Comment:
I'd remove this whole block. Just do the `ioutil.ReadDir` and check [`os.IsNotExist`](https://golang.org/pkg/os/#IsNotExist) on the error if you want special error handling in that case. For the `IsDir` check returning the error should still give us that information, we don't need to wrap it.

Reply:
Makes sense to me. Addressed.